controller/program/src/instructions/admin/add_lst.rs

```diff
@@ -0,0 +1,173 @@
+        .with_protocol_fee_accumulator_auth(&PROTOCOL_FEE_ID)
+        .with_pool_state(&POOL_STATE_ID)
+        .with_lst_state_list(&LST_STATE_LIST_ID)
+        .with_associated_token_program(&ATOKEN_ID)
+        .with_system_program(&SYS_PROG_ID)
+        .with_lst_token_program(token_prog)
+        // Free account - payer can be any account with sufficient lamports for ATA rent
+        .with_payer(abr.get(*accs.payer()).key())
+        // Free account - admin can specify any sol value calculator program
+        .with_sol_value_calculator(abr.get(*accs.sol_value_calculator()).key())
+        .build();
+
+    verify_pks(abr, &accs.0, &expected_pks.0)?;
+    verify_signers(abr, &accs.0, &ADD_LST_IX_IS_SIGNER.0)
+        .map_err(|expected_signer| log_and_return_acc_privilege_err(abr, *expected_signer))?;
+
+    verify_tokenkeg_or_22_mint(lst_mint_acc)?;
+    verify_sol_value_calculator_is_program(abr.get(*accs.sol_value_calculator()))?;
+
+    // Verify no duplicate in lst state list
+    let lst_state_list_acc = abr.get(*accs.lst_state_list());
+    let list = LstStatePackedList::of_acc_data(lst_state_list_acc.data())
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidLstStateListData))?;
+
+    if list.find_by_mint(lst_mint_acc.key()).is_some() {
+        return Err(Inf1CtlCustomProgErr(Inf1CtlErr::DuplicateLst).into());
+    }
+
+    verify_not_rebalancing_and_not_disabled(pool)?;
+
+    let is_lsl_uninitialized = lst_state_list_acc.data().is_empty();
+
+    // Create pool reserves and protocol fee accumulator ATAs if they do not exist
+    [
+        (*accs.pool_reserves(), *accs.pool_state()),
+        (
+            *accs.protocol_fee_accumulator(),
+            *accs.protocol_fee_accumulator_auth(),
+        ),
+    ]
+    .into_iter()
+    .try_for_each(|(ata, wallet)| -> Result<(), ProgramError> {
+        let create_accs = NewCreateIxAccsBuilder::start()
+            .with_funding(*accs.payer())
+            .with_ata(ata)
+            .with_wallet(wallet)
+            .with_mint(*accs.lst_mint())
+            .with_sys_prog(*accs.system_program())
+            .with_token_prog(*accs.lst_token_program())
+            .build();
+
+        cpi.invoke_fwd(
+            abr,
+            &ATOKEN_ID,
+            CreateIdempotentIxData::as_buf(),
+            create_accs.0,
+        )?;
+
+        Ok(())
+    })?;
+
+    // Realloc lst state list
+    if is_lsl_uninitialized {
+        assign_invoke_signed(
+            abr,
+            cpi,
+            NewAssignIxAccsBuilder::start()
+                .with_assign(*accs.lst_state_list())
+                .build(),
+            &ID,
+            &[PdaSigner::new(&[
+                PdaSeed::new(&LST_STATE_LIST_SEED),
+                PdaSeed::new(&[LST_STATE_LIST_BUMP]),
+            ])],
+        )?;
+    }
+
+    let lst_state_list_acc = abr.get_mut(*accs.lst_state_list());
+    lst_state_list_acc.grow_by(size_of::<LstStatePacked>(), false)?;
```

Comment:
there is some common code here that can be shared with the `DisablePoolAuthorityList`, i think its in https://github.com/igneous-labs/S/blob/813333ba8f02d2ff05afc1fcbf427a2210dda36a/programs/s-controller/src/list_account.rs in the old code. But lets do the refactor when we get to the `DisablePoolAuthorityList` part